implementation/README.md

```diff
@@ -0,0 +1,147 @@
+|`ME/TXE firmware version 2.0.5.3112 (generation 2)`|
+|`Public key match: Intel TXE, firmware versions 2.x.x.x`|
+|`The AltMeDisable bit is SET`|
+|`Setting the AltMeDisable bit in PCHSTRP10 to disable Intel ME…`|
+|`Checking the FTPR RSA signature... VALID`|
+|`Done! Good luck!`|
+
+## LinuxBoot Implementation steps
+
+### Step 1: boot Linux via netboot / UEFI shell
+
++   netboot: standard BIOS-based PXE boot
+
+    +   Netboot is probably the most common working boot method on UEFI
+    +   We have never seen a system that did not have a net boot
+
++   UEFI Shell (mentioned only for completeness)
+
+    +   Install Linux on FAT-32 media with a name of your choice (e.g. "kernel")
+        +   FAT-32, also known as MS-DOS file system
+    +   Boot kernel at UEFI Shell prompt
+    +   We've run into a few systems that don't have a UEFI shell
+
+#### Working with a system that only has a net interface
+
+If the system only has a net interface, you use Dynamic Host Configuration Protocol (DHCP), using broadcast DISCOVER, and Trivial File Transfer Protocol (TFTP) to get the boot information you need. These programs are provided by the PXEboot program.
+
+Configuration information is provided by REPLY to a DHCP request. The REPLY returns an IP, server, and a configuration file name that provides:
+
+   +   Identity
+   +   What to boot
+   +   Where to get it
+
+Data is provided by TFTP. On proper systems, HTTP downloading takes a fraction of a second even for 16M kernels. With TFTP it's very slow and TFTP won't work with initramfs much large than 32MiB. Most LinuxBoot shops use or are transitioning to HTTP.
+
+Note: Boot images require a kernel(bzImage) + an initramfs + a command line. They can be loaded as three pieces or compiled and loaded as one piece, as described in this section.
+
+### Step 2: replace Shell binary section
+
++   UEFI Shell is a DXE
+
+    +   DXEs are Portable Executable 32-bit binaries (PE32)
+    +   They have multiple sections, one of them being binary code
+    +   You need a flash image (in this case called firmware.bin). You can get it via vendor website, flashrom, or other mechanism.
+
++   The following `utk` command replaces the Shell code section with a Linux kernel:
+
+      ` utk firmware.bin replace_pe32 Shell bzImage save new.bin`
+
+     Note: it's always a PE32, even for 64-bit kernels
+
++   After running `utk`, you can reflash
+
+### Step 3: remove as many DXEs as possible
+
++   You can do an initial mass removal based on your current knowledge
++   `utk` automates removing DXEs: this is the DXE cleaner
+
+    +   `utk` removes a DXE, reflashes, checks if it boots, repeats
+
+    This part should be easy: DXE can have a dependency section. In practice, it's hard: because dependency sections are full of errors and omissions. Sloppy UEFI code frequently does not check for failed DXE loads, e.g., remove EnglishDXE, get a General Protection Fault.
```

Comment:
Sloppy -> A lot of

Reply:
can remove frequently

Reply:
DXE loads. -> full stop. you can remove the e.g.

Reply:
I made these three changes.